### PR TITLE
CLI support for max-width parameter

### PR DESCRIPTION
**Pull request by lukas-krecan**

### CODE REVIEW

**Comment by laurentthiebaudezm** [comment]
@lukas-krecan have you any news? Any reviewer to add ?

Interesting for us also :pray:

**Comment by lukas-krecan** [comment]
Unfortunately no 🤷‍♂️

**Comment by hick209** [comment]
I'll kick start an internal discussion about this

**Comment by davidtorosyan** [comment]
Thanks for the submission, and sorry for the late response. Also sorry to say this is not something we're likely to include.

As for why, ktfmt inherits a lot of its code (and philosophy) from google-java-format, so I'll point to the discussion in [this comment](https://github.com/google/google-java-format/pull/57#issuecomment-388125762), relevant bit here:
> The explicit goals we had for building this tool are to bring consistency to Java code, and to free developers from arguments over style choices. It has always been an explicit non-goal to support developers' individual preferences, because doing so actually works directly against the primary goals.

Basically, we believe exposing such an option would be more trouble than it's worth, not just for maintenance but also usability.

**Comment by lukas-krecan** [comment]
This is confusing. The option is available through the Gradle plugin and most likely through other plugins too. The PR just makes an existing option available from the CLI. Could you please explain why CLI should have less options than Gradle plugin?

**Comment by davidtorosyan** [comment]
> The option is available through the Gradle plugin and most likely through other plugins too.

That might be a bug. I'm looking at the code and I don't see how that's configured. Mind pointing me to the setting if you know where it is?

File: core/src/main/java/com/facebook/ktfmt/cli/ParsedArgs.kt [hunk 1]
```diff
@@ -87,5 +92,12 @@ data class ParsedArgs(
       }
       return parts[1]
     }
+
+    private fun String?.asInt(err: PrintStream, name: String): Int? = try {
+      this?.toInt()
+    } catch (e: NumberFormatException) {
+      err.println("Invalid $name: $this")
+      null
+    }
```
[the remark this answers is not in the document]
Comment (lukas-krecan) [reply core/src/main/java/com/facebook/ktfmt/cli/ParsedArgs.kt]: Thanks, I have updated the code accordingly.

**Pull request closed without merging**